.github/workflows/ci.yaml

```diff
@@ -2303,6 +2706,56 @@ jobs:
       - name: ghcr.io/hyperledger/cactus-quorum-multi-party-all-in-one
         run: DOCKER_BUILDKIT=1 docker build ./tools/docker/quorum-multi-party-all-in-one/ -f ./tools/docker/quorum-multi-party-all-in-one/Dockerfile -t cactus-quorum-multi-party-all-in-one
     
+      - name: Run Trivy vulnerability scan for cactus-quorum-multi-party-all-in-one
+        uses: aquasecurity/trivy-action@0.11.2
+        with:
+          image-ref: 'cactus-quorum-multi-party-all-in-one'
+          format: 'table'
+          exit-code: '1'
+          ignore-unfixed: true
+          vuln-type: 'os,library'
+          severity: 'CRITICAL,HIGH'
+  generate_coverage_report:
+    runs-on: ubuntu-22.04
+    needs: [check-coverage, cp-bungee-hermes]
```

Comment:
@aldousalvarez Why does this specifically depend on `cp-bungee-hermes` what makes that package special compard to all the other packages we are also testing in this workflow?

Reply:
@petermetz I chose cp-bungee-hermes because when we try to generate packages it is running simultaneously on all the step when they run tools.ci.sh it has the longest time that is passing when running tools.ci.sh because we need to make sure all the packages should generate a coverage report first before we merged them all because when I tried to depend on check-coverage only some of the packages does not get included only the finished ones are being covered by the code coverage report.

Reply:
If the final job was in a separate workflow triggered from ci.yaml, then you wouldn't need to worry about a longer test job being introduced.  

https://stackoverflow.com/questions/58457140/dependencies-between-workflows-on-github-actions

While I'm new to github, I get the impression artifacts are shared with a workflow triggered in sequence if they are uploaded by the upstream workflow.

https://stackoverflow.com/questions/60355925/share-artifacts-between-workflows-github-actions

Reply:
@jenniferlianne oh that it nice. I will take a look on this one and try it out. Thanks

Reply:
@aldousalvarez 

> @petermetz I chose cp-bungee-hermes because when we try to generate packages it is running simultaneously on all the step when they run tools.ci.sh it has the longest time that is passing when running tools.ci.sh because we need to make sure all the packages should generate a coverage report first before we merged them all because when I tried to depend on check-coverage only some of the packages does not get included only the finished ones are being covered by the code coverage report.

@aldousalvarez Unfortunately, that is not an optimal solution because the time each job takes in relation to each other is too varied on account of GitHub being a multi-tenant application with shared CPUs between jobs of ours and other people working on other projects. There are also a lot of other factors in play that changes (basically randomizes) the time jobs take to execute such as network congestion while fetching dependencies. For the CPU usage part, you can read more about the theory behind it if you look up the noisy neighbors[1] problem for example.

Long story short: I agree with @jenniferlianne 's excellent suggestion, please refactor it so that the ci.yaml workflow's conclusion triggers the code coverage which can be in a separate workflow file that could be named just `CI` and the job within it could be called `coverage_typescript`. We need to specify that it's typescript coverage because in the future we plan on having coverage for all the different programming languages that are present in the mono-repo (Kotlin, Rust, Go, Solidity etc.)

[1] https://learn.microsoft.com/en-us/azure/architecture/antipatterns/noisy-neighbor/noisy-neighbor

Reply:
@petermetz Okay, Got it. Will do it and I am currently in progress of refactoring the code and testing how to integrate it. Thank you